pkg/download/protocol.go

```diff
@@ -256,12 +257,12 @@ func (p *protocol) processDownload(ctx context.Context, mod, ver string, f func(
 		}
 		return f(newVer)
 	case mode.Async:
-		go p.stasher.Stash(ctx, mod, ver)
+		go p.stasher.Stash(config.ContextWithoutCancel(ctx), mod, ver)
```

Comment:
I think I understand what is happening here. Do you think it is better to create a new context and copy all the Values in the new context? 

The only value that is being set is the request-id here:

```
const HeaderKey = "Athens-Request-ID"
```

https://github.com/gomods/athens/blob/0c78ebd6308329713a2d2dd9bc007ca9c36d98cc/pkg/requestid/requestid.go

Reply:
Agreed. It's better to just make a new context right there and copy the request-id over.

Reply:
Actually the other value being set that might is the log entry which would be good to include to keep the same contextual information: https://github.com/gomods/athens/blob/main/pkg/log/log_context.go

Reply:
My change was proposed to avoid any hard to track bugs with missing copied ctx values or deadlines being set before spinning up the go-routine - trying to copy over deadlines and ctx values could lead to some hard to track issues

Example of ctx values needed to copy over as well is whatever each observability tool use to track trace and spans

Example of deadline is whatever the http server is/could be configured with to avoid long hanging requests using up resources

Reply:
@jippi Thanks for the explanation. I am okay with this change. 
I think async processes should have their own deadlines but thisis okay!

Marwan wrote most of this bit; so as long as he is okay. LGTM